crates/ironrdp-pdu/src/utils.rs

```diff
@@ -248,3 +250,44 @@ impl<T> SplitTo for &mut [T] {
         a
     }
 }
+
+pub trait CheckedAdd: Sized + Add<Output = Self> {
+    fn checked_add(self, rhs: Self) -> Option<Self>;
+}
+
+// Implement the trait for usize and u32
+impl CheckedAdd for usize {
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        usize::checked_add(self, rhs)
+    }
+}
+
+impl CheckedAdd for u32 {
+    fn checked_add(self, rhs: Self) -> Option<Self> {
+        u32::checked_add(self, rhs)
+    }
+}
+
+// Utility function for checked addition that returns a PduResult
+pub fn checked_sum<T>(values: &[T]) -> PduResult<T>
+where
+    T: CheckedAdd + Copy + Debug,
+{
+    values.split_first().map_or_else(
+        || Err(other_err!("Empty array provided to checked_sum")),
+        |(&first, rest)| {
+            rest.iter().try_fold(first, |acc, &val| {
+                acc.checked_add(val)
+                    .ok_or_else(|| other_err!("Overflow detected during addition"))
+            })
+        },
+    )
+}
+
+// Utility function that panics on overflow
+pub fn checked_sum_or_panic<T>(values: &[T]) -> T
```

Comment:
naming: `checked_` prefix is used when the API does not panic (e.g.: `checked_add`). [Panicking API are using the `strict_` prefix](https://github.com/rust-lang/rust/issues/118260) (e.g.: `strict_add`). API which may or may not panic depending on compilation flags does not have any prefix (e.g.: `add`). In that sense, `checked_sum_or_panic` is a bit alien. I suggest `strict_sum`.